File: reverse_list.py
class Node:
    def __init__(self, v):
        self.v = v
        self.next = None


def reverse_list(root):
    p = root
    head = Node(0)
    while p is not None:
        p_next = p.next
        p.next = head.next
        head.next = p
        p = p_next
    return head.next


if __name__ == "__main__":
    list1 = Node(1)
    list1.next = Node(2)
    list1.next.next = Node(3)

    reversed_list1 = reverse_list(list1)
    print(reversed_list1.v)
    print(reversed_list1.next.v)
    print(reversed_list1.next.next.v)

File: go/reverse_list.go
package main

import (
	"fmt"
)

type Node struct {
	v    int
	next *Node
}

func reverseList(root *Node) *Node {
	p := root
	head := &Node{v: 0, next: nil}
	for p != nil {
		p_next := p.next
		p.next = head.next
		head.next = p
		p = p_next
	}
	return head.next
}

func main() {
	list1 := &Node{v: 1, next: &Node{v: 2, next: &Node{v: 3, next: nil}}}
	list2 := reverseList(list1)
	fmt.Printf("%+v", list2)
}
